website/src/components/discourse/index.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from 'react'
+import axios from 'axios'
+import feedStyles from './styles.module.css';
+
+export default function DiscourseFeed({
+  order = 'latest_topic',
+  status = undefined,
+  after = undefined,
+  before = undefined,
+  inString = undefined, 
+  min_posts = undefined,
+  max_posts = undefined,
+  min_views = undefined,
+  max_views = undefined,
+  tags = undefined, 
+  term = undefined,
+  category = undefined,
+  title = undefined,
+  link_text = "See latest topics",
+  link_href = "https://discourse.getdbt.com/",
+  post_count = 5,
+  styles = {}
+}) {
+
+  const [posts, setPosts] = useState([])
+  const [loading, setLoading] = useState(true)
+  const [isError, setIsError] = useState(false)
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        // Ensure error state is false and loading true
+        setLoading(true)
+        setIsError(false)
+
+        // Build Netlify Function endpoint
+        const endpoint = window?.location?.hostname?.includes('localhost')
+          ? 'http://localhost:8888/.netlify/functions/get-discourse-posts'
+          : '/.netlify/functions/get-discourse-posts'
+
+        // Get Discourse topics data
+        const { data } = await axios.post(endpoint, {
+          status,
+          order,
+          after,
+          before,
+          inString, 
+          min_posts,
+          max_posts,
+          min_views,
+          max_views,
+          tags, 
+          term,
+          category,
+        })
+
+        // Set error state if data not available
+        if(!data) throw new Error('Unable to get latest topics.')
+
+        // Set 5 latest posts
+        setPosts(data.slice(0, post_count))
+        setLoading(false)
+      } catch(err) {
+        setIsError(true)
+        setLoading(false)
+      }
+    }
+    fetchData()
+  }, [])
+
+  // Set initial min-height
+  // This is to avoid layout shifts
+  // which affects Lighthouse performance scores
+  const setMinHeight = isError
+    ? 'auto'
+    : 414
+
+  return (
+    <div className={feedStyles.discoursePosts} style={{minHeight: setMinHeight, ...styles}}>
+      {title && (
+        <h2>{title}</h2>
+      )}
+      {loading ? (
+        <img src="/img/loader-icon.svg" alt="Loading" className={feedStyles.loadingIcon} />
+      ) : isError || !posts?.length > 0 ? (
+        <p>Unable to load Discourse posts at this time.</p>
```

Comment:
```suggestion
        <p>Unable to load forum posts at this time.</p>
```

I think we'll de-emphasise the platform we're using and lean into it just being the Forum (in the UI at least; the code can still be specific)